x-pack/plugins/ml/public/explorer/explorer.js

```diff
@@ -900,10 +907,44 @@ export const Explorer = injectI18n(injectObservablesAsProps(
       }
     }
 
+    applyFilterFromTable = (fieldName, fieldValue, action) => {
+      let newQueryString = '';
+      const { queryString } = this.state;
+      const operator = 'and ';
+
+      if (action === '+') {
+        // Don't re-add if already exists in the query
+        const queryPattern = getQueryPattern(fieldName, fieldValue);
+        if (queryString.match(queryPattern) !== null) {
+          return;
+        }
+        newQueryString = `${queryString ? `${queryString} ${operator}` : ''}${fieldName}:"${fieldValue}"`;
+      } else if (action === '-') {
+        if (this.state.filterActive === false) {
+          return;
+        } else {
+          newQueryString = removeFilterFromQueryString(this.state.queryString, fieldName, fieldValue);
+        }
+      }
+
+      try {
+        const queryValues = getKqlQueryValues(`${newQueryString}`, this.state.indexPattern);
+        this.applyInfluencersFilterQuery(queryValues);
+      } catch(e) {
+        console.log('Invalid kuery syntax', e); // eslint-disable-line no-console
+
+        toastNotifications.addDanger(this.props.intl.formatMessage({
+          id: 'xpack.ml.explorer.invalidKuerySyntaxErrorMessageFromTable',
+          defaultMessage: 'Uh oh! Looks like the filter syntax is invalid. Please check your input and try again.'
```

Comment:
As discussed, I'd change this message to something like `Invalid syntax in query bar. The input must be valid Kibana Query Language (KQL).` Could you add a link to the KQL page in the docs?